Sources/AsyncHTTPClient/ConnectionPool.swift

```diff
@@ -61,69 +62,46 @@ final class ConnectionPool {
     /// - Returns: A connection  corresponding to the specified parameters
     ///
     /// When the pool is asked for a new connection, it creates a `Key` from the url associated to the `request`. This key
-    /// is used to determine if there already exists an associated `HTTP1ConnectionProvider` in `connectionProviders`.
+    /// is used to determine if there already exists an associated `HTTP1ConnectionProvider` in `providers`.
     /// If there is, the connection provider then takes care of leasing a new connection. If a connection provider doesn't exist, it is created.
-    func getConnection(for request: HTTPClient.Request, preference: HTTPClient.EventLoopPreference, on eventLoop: EventLoop, deadline: NIODeadline?) -> EventLoopFuture<Connection> {
+    func getConnection(for request: HTTPClient.Request, preference: HTTPClient.EventLoopPreference, on eventLoop: EventLoop, deadline: NIODeadline?, setupComplete: EventLoopFuture<Void>) -> EventLoopFuture<Connection> {
         let key = Key(request)
 
-        let provider: HTTP1ConnectionProvider = self.connectionProvidersLock.withLock {
-            if let existing = self.connectionProviders[key] {
-                existing.stateLock.withLock {
-                    existing.state.pending += 1
-                }
+        let provider: HTTP1ConnectionProvider = self.lock.withLock {
+            if let existing = self.providers[key], existing.enqueue() {
                 return existing
             } else {
-                let http1Provider = HTTP1ConnectionProvider(key: key, eventLoop: eventLoop, configuration: self.configuration, parentPool: self)
-                self.connectionProviders[key] = http1Provider
-                http1Provider.stateLock.withLock {
-                    http1Provider.state.pending += 1
-                }
-                return http1Provider
+                let provider = HTTP1ConnectionProvider(key: key, eventLoop: eventLoop, configuration: self.configuration, pool: self)
+                self.providers[key] = provider
```

Comment:
Are we not missing a call to `enqueue` here?

Reply:
Ok, after reading through the entire PR I finally found out why we aren't. Can we add a comment here that states that providers begin at `+1`?

Reply:
Done!